### PR TITLE
Feature new sdk implement download tracker data

### LINKED ISSUE #272: Implement download tracker data

### PR DESCRIPTION
**Pull request by xurxodev**
close #272

### CODE REVIEW

File: core/src/androidTest/java/org/hisp/dhis/android/core/event/EventEndPointCallMockIntegrationShould.java [hunk 1]
```diff
@@ -56,12 +56,12 @@ public void tearDown() throws IOException {
     public void download_events_according_to_default_query() throws Exception {
```
Comment (ifoche) [inline core/src/androidTest/java/org/hisp/dhis/android/core/event/EventEndPointCallMockIntegrationShould.java:56]: @xurxodev I'm getting a FOREIGN KEY constraint failed error in this test. Is it working for you?

File: core/src/main/java/org/hisp/dhis/android/core/trackedentity/TrackedEntityAttributeValueHandler.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        for (TrackedEntityAttributeValue attributeValue : attributeValuesInDB) {
+            if (!existsDataElement(attributeValues, attributeValue.trackedEntityAttribute())) {
+                attributeUIdsToRemove.add(attributeValue.trackedEntityAttribute());
+            }
+        }
+
+        return attributeUIdsToRemove;
+    }
+
+    private boolean existsDataElement(List<TrackedEntityAttributeValue> attributeValues,
```
Comment (ifoche) [inline core/src/main/java/org/hisp/dhis/android/core/trackedentity/TrackedEntityAttributeValueHandler.java]: @xurxodev didn't you mean something like existsTrackedEntityAttribute here? Looking at the code, it looks like it's checking if the TE attributes exist for a list of TE attributeValues. Is that it?

**Review by ifoche — CHANGES_REQUESTED** [review]
@xurxodev I'm asking only for a couple of minor changes/questions

File: core/src/androidTest/java/org/hisp/dhis/android/core/event/EventEndPointCallMockIntegrationShould.java (hunk 1 above)
Comment (xurxodev) [reply core/src/androidTest/java/org/hisp/dhis/android/core/event/EventEndPointCallMockIntegrationShould.java:56]: @ifoche All tests is working for me

File: core/src/main/java/org/hisp/dhis/android/core/trackedentity/TrackedEntityAttributeValueHandler.java (hunk 2 above)
Comment (xurxodev) [reply core/src/main/java/org/hisp/dhis/android/core/trackedentity/TrackedEntityAttributeValueHandler.java]: @ifoche yes sorry, It's a copy and paste from events solution, I have renamed these method

**Comment by xurxodev** [comment]
@ifoche I have done the changes and answered your questions

File: core/src/androidTest/java/org/hisp/dhis/android/core/event/EventEndPointCallMockIntegrationShould.java (hunk 1 above)
Comment (ifoche) [reply core/src/androidTest/java/org/hisp/dhis/android/core/event/EventEndPointCallMockIntegrationShould.java:56]: @xurxodev not sure of what was it but now it also works for me

**Pull request merged**